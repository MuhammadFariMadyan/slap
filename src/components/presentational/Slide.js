import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { MdClose } from "react-icons/md";
import "typeface-roboto";

import Thumbnail from "./Thumbnail";

const Container = styled.div`
  width: 200px;
  background: ${({ isActive }) => (isActive ? "#ddd" : "#eee")};
  margin-right: 4px;
  display: flex;
  justify-content: flex-end;
  padding: 4px 8px;
  box-sizing: border-box;
`;

const Index = styled.p`
  font-size: 12px;
  font-family: Roboto;
  font-weight: bold;
  color: #333;
  margin-top: 4px;
  margin-right: 8px;
`;

export const IconClose = styled(MdClose)`
  font-size: 14px;
  color: #aaa;
  margin-left: 8px;
  cursor: pointer;
`;

const Slide = ({
  className,
  index,
  contents,
  onClick,
  onClickDelete,
  isActive
}) => (
  <Container className={className} isActive={isActive}>
    <Index>{index}</Index>
    <Thumbnail width={128} contents={contents} onClick={onClick} />
    <IconClose onClick={onClickDelete} />
  </Container>
);

Slide.propTypes = {
  className: PropTypes.string,
  index: PropTypes.number,
  contents: PropTypes.array,
  onClick: PropTypes.func,
  onClickDelete: PropTypes.func,
  isActive: PropTypes.bool
};

export default Slide;
