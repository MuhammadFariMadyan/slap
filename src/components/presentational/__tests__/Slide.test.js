import React from "react";
import { shallow } from "enzyme";
import "jest-styled-components";

import Slide, { IconClose } from "../Slide";

jest.mock("../Thumbnail", () => "Thumbnail");

describe("Slide", () => {
  const setup = propOverrides => {
    const props = {
      index: 1,
      contents: [
        {
          id: "content-1",
          slideID: "slide-1",
          projectID: "project-1",
          type: "text",
          content: "Hello World",
          position: {
            x: 32,
            y: 32
          }
        },
        {
          id: "content-2",
          slideID: "slide-1",
          projectID: "project-1",
          type: "text",
          content: "Stay hungry",
          position: {
            x: 128,
            y: 96
          }
        }
      ],
      onClick: jest.fn(),
      onClickDelete: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Slide {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule("background", "#eee");
  });

  it("should renders in active", () => {
    const { wrapper } = setup({ isActive: true });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule("background", "#ddd");
  });

  it("should calls 'onClick' callback while clicked", () => {
    const { wrapper, props } = setup();

    wrapper.find("Thumbnail").simulate("click");

    expect(props.onClick).toBeCalled();
  });

  it("should calls 'onClickDelete' callback while delete clicked", () => {
    const { wrapper, props } = setup();

    wrapper.find(IconClose).simulate("click");

    expect(props.onClickDelete).toBeCalled();
  });
});
