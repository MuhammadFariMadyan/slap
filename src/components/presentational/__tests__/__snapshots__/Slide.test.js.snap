// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide should renders default 1`] = `
<styled.div>
  <styled.p>
    1
  </styled.p>
  <Thumbnail
    contents={
      Array [
        Object {
          "content": "Hello World",
          "id": "content-1",
          "position": Object {
            "x": 32,
            "y": 32,
          },
          "projectID": "project-1",
          "slideID": "slide-1",
          "type": "text",
        },
        Object {
          "content": "Stay hungry",
          "id": "content-2",
          "position": Object {
            "x": 128,
            "y": 96,
          },
          "projectID": "project-1",
          "slideID": "slide-1",
          "type": "text",
        },
      ]
    }
    onClick={[Function]}
    width={128}
  />
  <Styled(Component)
    onClick={[Function]}
  />
</styled.div>
`;

exports[`Slide should renders in active 1`] = `
<styled.div
  isActive={true}
>
  <styled.p>
    1
  </styled.p>
  <Thumbnail
    contents={
      Array [
        Object {
          "content": "Hello World",
          "id": "content-1",
          "position": Object {
            "x": 32,
            "y": 32,
          },
          "projectID": "project-1",
          "slideID": "slide-1",
          "type": "text",
        },
        Object {
          "content": "Stay hungry",
          "id": "content-2",
          "position": Object {
            "x": 128,
            "y": 96,
          },
          "projectID": "project-1",
          "slideID": "slide-1",
          "type": "text",
        },
      ]
    }
    onClick={[Function]}
    width={128}
  />
  <Styled(Component)
    onClick={[Function]}
  />
</styled.div>
`;
